tests/Unit/Domain/Test_OrientationMap.cpp

```diff
@@ -259,3 +263,181 @@ SPECTRE_TEST_CASE("Unit.Domain.OrientationMap", "[Domain][Unit]") {
+#endif
+}
+
+// [[OutputRegex, This OrientationMap fails to map Directions one-to-one.]]
+[[noreturn]] SPECTRE_TEST_CASE("Unit.Domain.OrientationMap.BijectiveHost",
+                               "[Domain][Unit]") {
+  ASSERTION_TEST();
+#ifdef SPECTRE_DEBUG
+  auto failed_orientationmap = OrientationMap<2>{
+      std::array<Direction<2>, 2>{
+          {Direction<2>::upper_xi(), Direction<2>::lower_xi()}},
+      std::array<Direction<2>, 2>{
+          {Direction<2>::upper_xi(), Direction<2>::upper_eta()}},
+  };
+  static_cast<void>(failed_orientationmap);
+
+  ERROR("Failed to trigger ASSERT in an assertion test");
+#endif
+}
+
+// [[OutputRegex, This OrientationMap fails to map Directions one-to-one.]]
+[[noreturn]] SPECTRE_TEST_CASE("Unit.Domain.OrientationMap.BijectiveNeighbor",
+                               "[Domain][Unit]") {
+  ASSERTION_TEST();
+#ifdef SPECTRE_DEBUG
+  auto failed_orientationmap = OrientationMap<3>{
+      std::array<Direction<3>, 3>{{Direction<3>::upper_xi(),
+                                   Direction<3>::lower_eta(),
+                                   Direction<3>::lower_zeta()}},
+      std::array<Direction<3>, 3>{{Direction<3>::upper_xi(),
+                                   Direction<3>::upper_eta(),
+                                   Direction<3>::lower_eta()}},
+  };
+  static_cast<void>(failed_orientationmap);
+
+  ERROR("Failed to trigger ASSERT in an assertion test");
+#endif
+}
+
+namespace {
+template <size_t VolumeDim>
+OrientationMap<VolumeDim> generate_orientation_map(
+    gsl::not_null<VolumeCornerIterator<VolumeDim>*> vci,
+    size_t num_of_cycle_permutation) {
+  std::array<size_t, VolumeDim> dimensions{};
+  std::iota(dimensions.begin(), dimensions.end(), 0);
+  std::array<Direction<VolumeDim>, VolumeDim> array_of_directions{};
+  for (size_t i = 0; i < num_of_cycle_permutation; i++) {
+    std::next_permutation(dimensions.begin(), dimensions.end());
+  }
+  for (size_t i = 0; i < VolumeDim; i++) {
+    gsl::at(array_of_directions, i) =
+        Direction<VolumeDim>{gsl::at(dimensions, i), gsl::at(vci->sides(), i)};
+  }
+  return OrientationMap<VolumeDim>{array_of_directions};
+}
+}  // namespace
+
+SPECTRE_TEST_CASE("Unit.Domain.DiscreteRotation.AllOrientations",
+                  "[Domain][Unit]") {
+  for (size_t permutation = 0; permutation < factorial(2); permutation++) {
+    for (VolumeCornerIterator<2> vci{}; vci; ++vci) {
+      OrientationMap<2> map_2d = generate_orientation_map<2>(&vci, permutation);
+      const std::array<double, 2> original_point{{0.5, -2.0}};
+      const std::array<double, 2> new_point =
+          discrete_rotation(map_2d, original_point);
+      for (size_t d = 0; d < 2; d++) {
+        CHECK(gsl::at(new_point, d) ==
+              (map_2d(Direction<2>{d, Side::Upper}).side() == Side::Upper
+                   ? gsl::at(original_point, map_2d(d))
+                   : -1.0 * gsl::at(original_point, map_2d(d))));
```

Comment:
I'm not going to make you change it, but my original thought here was to just test the basis vectors (and maybe their negatives).  Then it is easy to get the correct expected value from the OrientationMap without having to worry about all this permutation logic (it is just map the direction and have a function to take Direction -> array).  That keeps it further from the implementation of the function itself.  It is less thorough, but you already have a test of applying the function to generic arrays..